### PR TITLE
adapt to new y2base argument passing

### PR DESCRIPTION
**Pull request by jreidinger**
will fail, as it use y2base from yast2-core and not one from ruby-bindings
fail will be solved by dependencies.

### CODE REVIEW

**Comment by mvidner** [comment]
da74e25 "WIP of rewritting DnsServerAPI to ruby" does not belong to this PR

**Comment by jreidinger** [comment]
damn, you are right, it is part of pr to drop perl. I will remove it.

File: scripts/yast2 [hunk 1]
```diff
@@ -285,19 +285,14 @@ else
...
 
 if [ -n "$CMDLINE_HELP" ]; then
     set -- $@ help;
 fi
 if [ $# -gt 0 ]; then
     a=("$@")
-    b=("${a[@]/#/(}")
-    c=("${b[@]/%/)}")
-    set -- -S "${c[@]}"
+    set -- --arg "${a[@]}"
```
Comment (mvidner) [inline scripts/yast2]: This is wrong. `-S` starts multiple arguments but `--arg` has a single argument.

Comment (mvidner) [inline scripts/yast2]: BTW you can debug this with
```sh
printf "%q\n" "${@}"
printf "%q\n" "${a[@]}"
```

File: scripts/yast2 [hunk 2]
```diff
@@ -290,10 +290,11 @@ fi
...
 fi
-if [ $# -gt 0 ]; then
-    a=("$@")
-    set -- --arg "${a[@]}"
-fi
+ARGS=""
+for i in $@; do
+  ARGS="$ARGS --arg $i"
+done
+set -- $ARGS
```
Comment (mvidner) [inline scripts/yast2]: No, this breaks arguments that contain spaces. Please test that.

File: scripts/yast2 [hunk 3]
```diff
@@ -290,10 +290,11 @@ fi
 if [ -n "$CMDLINE_HELP" ]; then
     set -- $@ help;
 fi
-if [ $# -gt 0 ]; then
-    a=("$@")
-    set -- --arg "${a[@]}"
-fi
+ARGS=""
+for i in "$@"; do
+  ARGS="$ARGS --arg '$i'"
```
Comment (mvidner) [inline scripts/yast2]: This will still fail if the value contains a single quote

Comment (jreidinger) [inline scripts/yast2]: so any suggestion to use this ****** shell **** programming language?

Comment (jreidinger) [inline scripts/yast2]: and also what is probability that some parameter will contain it?

Comment (mvidner) [inline scripts/yast2]: We got a bug once, not for a quote but for a backslash: https://bugzilla.suse.com/show_bug.cgi?id=382883

Comment (mvidner) [inline scripts/yast2]: Bash is awful, but it does have a solution for this: `printf %q` is the proper way to quote for an `eval`:

```sh
ARGS=$(printf '%s -arg %q' "$ARGS" "$i")
```

File: scripts/yast2 [hunk 4]
```diff
@@ -285,20 +284,16 @@ else
...
 fi
-if [ $# -gt 0 ]; then
-    a=("$@")
-    b=("${a[@]/#/(}")
-    c=("${b[@]/%/)}")
-    set -- -S "${c[@]}"
-fi
+ARGS=""
+for i in "$@"; do
+  ARGS="$ARGS --arg `printf %q "$i"`"
```
Comment (mvidner) [inline scripts/yast2]: Well, my earlier suggestion, and this variant, are both wrong.
We need tests for this.

Comment (jreidinger) [inline scripts/yast2]: I testing it manually trying various stuff on CLI. Why is this wrong? And for tests you mean tests in bash? I hope not

**Comment by mvidner** [comment]
When I take today's commits and apply them to master, the test should pass. But it still fails, and it is hard to see why:
```
[   61s] Failures:
[   61s] 
[   61s]   1) yast2 script pass properly all arguments
[   61s]      Failure/Error: Cheetah.run(TEST_DIR + "/yast2", "args_test_client", 'abc"\'\\|;&<>! ', "second", env: { "TESTING_YAST2" => "1" })
[   61s] 
[   61s]      Cheetah::ExecutionFailed:
[   61s]        Execution of "/home/abuild/rpmbuild/BUILD/yast2-3.2.14/scripts/yast2 args_test_client abc\"\'\\\|\;\&\<\>\!\  second" failed with status 1 (no error output).
[   61s]      # ./yast2_test.rb:13:in `block (2 levels) in <top (required)>'
```

**Comment by mvidner** [comment]
BTW when Cheetah says " (no error output)" it is because the program does
```sh
echo fail
exit 1
```
instead of
```sh
echo fail >&2
exit 1
```
and it cannot be helped with `stdout: :capture` or `stderr: :capture`.

**Comment by jreidinger** [comment]
@mvidner true, cheetah show in exception message only error ouput, but have access to both in exception - https://github.com/openSUSE/cheetah/blob/master/lib/cheetah.rb#L53

**Pull request merged**